### ISSUE TITLE
"Open Folder" option is not opening dialog box, also hanging whole system

### ISSUE THREAD

**Issue by myashjain10** [microsoft/vscode#230878]
Type: <b>Bug</b>

Under the File menu, Clicking any option that opens a "kdialog" like "Open Folder", "Open File", doesnt opens the dialog box for choosing the folder/file. When i check the processes running in System monitor, i can see "kdialog" running and its memory usage keeps increasing. After certain point the whole system hangs ( probably due to memory overload).

VS Code version: Code 1.94.1 (e10f2369d0d9614a452462f2e01cdc4aa9486296, 2024-10-05T05:44:32.189Z)
OS version: Linux x64 6.8.0-45-generic snap
Modes:

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i3-6006U CPU @ 2.00GHz (4 x 2000)|
|GPU Status|2d_canvas: unavailable_software<br>canvas_oop_rasterization: disabled_off<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: disabled_software<br>multiple_raster_threads: enabled_on<br>opengl: disabled_off<br>rasterization: disabled_software<br>raw_draw: disabled_off_ok<br>skia_graphite: disabled_off<br>video_decode: disabled_software<br>video_encode: disabled_software<br>vulkan: disabled_off<br>webgl: unavailable_software<br>webgl2: unavailable_software<br>webgpu: disabled_off<br>webnn: disabled_off|
|Load (avg)|2, 3, 2|
|Memory (System)|7.67GB (2.86GB free)|
|Process Argv|--no-sandbox --force-user-env --crash-reporter-id 6e9a9a44-03a3-4efb-bfeb-8bc4b1e5f65c|
|Screen Reader|no|
|VM|0%|
|DESKTOP_SESSION|plasma|
|XDG_CURRENT_DESKTOP|KDE|
|XDG_SESSION_DESKTOP|KDE|
|XDG_SESSION_TYPE|x11|
</details><details><summary>Extensions (22)</summary>

Extension|Author (truncated)|Version
---|---|---
vscode-tailwindcss|bra|0.12.11
es7-react-js-snippets|dsz|4.4.3
prettier-vscode|esb|11.0.0
auto-close-tag|for|0.5.15
auto-complete-tag|for|0.1.0
auto-rename-tag|for|0.1.10
code-runner|for|0.12.2
git-graph|mhu|1.30.0
debugpy|ms-|2024.10.0
python|ms-|2024.16.0
vscode-pylance|ms-|2024.10.1
cpptools|ms-|1.21.6
cpptools-extension-pack|ms-|1.3.0
java|red|1.35.1
intellicode-api-usage-examples|Vis|0.2.8
vscodeintellicode|Vis|1.3.1
vscode-gradle|vsc|3.16.4
vscode-java-debug|vsc|0.58.0
vscode-java-dependency|vsc|0.24.0
vscode-java-pack|vsc|0.29.0
vscode-java-test|vsc|0.42.0
vscode-maven|vsc|0.44.0

(2 theme extensions excluded)

</details><details>
<summary>A/B Experiments</summary>

```
vsliv368cf:30146710
vspor879:30202332
vspor708:30202333
vspor363:30204092
vscod805cf:30301675
binariesv615:30325510
vsaa593:30376534
py29gd2263:31024239
c4g48928:30535728
azure-dev_surveyone:30548225
962ge761:30959799
pythongtdpath:30769146
pythonnoceb:30805159
asynctok:30898717
pythonmypyd1:30879173
h48ei257:31000450
pythontbext0:30879054
accentitlementst:30995554
dsvsc016:30899300
dsvsc017:30899301
dsvsc018:30899302
cppperfnew:31000557
dsvsc020:30976470
pythonait:31006305
dsvsc021:30996838
01bff139:31013167
a69g1124:31058053
dvdeprecation:31068756
dwnewjupyter:31046869
impr_priority:31102340
nativerepl1:31139838
refactort:31108082
pythonrstrctxt:31112756
flightc:31134773
wkspc-onlycs-t:31132770
wkspc-ranged-t:31151552
cf971741:31144450
autoexpandse:31146404
iacca1:31150322
notype1cf:31151524
5fd0e150:31146321

```

</details>

**Comment by deepak1556** [comment]
Merging to https://github.com/microsoft/vscode/issues/230679

**Comment by denis-trofimov** [comment]
Hi Thanks!

I've observed on my system KDE Ubuntu 24.04 LTE this bug. 
Then in the VS code, I open a dialog window, it doesn't work; the kdialog process remains in my processes list. 

What is the VS code snap version that contains your fix?


In case somebody follows my steps, I got now this version, 169.

`snap refresh code --revision 169`
I tried to rollback from 172 to 171, 170—they are all broken for me.
